src/components/PostDetails.tsx

```diff
@@ -2,99 +2,38 @@ import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { CommentData } from '../types/Comment';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  initComments,
+  removeComment,
+  addComment as add,
+} from '../features/comments';
+
+export const PostDetails: React.FC = () => {
+  const post = useAppSelector(state => state.selectedPost);
+  const postId = post?.id || 0;
+  const { comments, loaded, hasError } = useAppSelector(
+    state => state.comments,
+  );
+  const dispatch = useAppDispatch();
   const [visible, setVisible] = useState(false);
```

Comment:
do we really need this state?

Reply:
Yes, we do) It opens form for adding a new comment and closes it when we load new post's comments